pol-core/pol/regions/miscrgn.cpp

```diff
@@ -124,65 +123,29 @@ void WeatherDef::copy_default_regions()
   for ( auto& realmregion : regionrealms )
   {
     Realms::Realm* realm = realmregion.first;
-    unsigned int gridwidth = realm->width() / ZONE_SIZE;
-    unsigned int gridheight = realm->height() / ZONE_SIZE;
-
-    for ( unsigned int i = 0; i < gridwidth; i++ )
-    {
-      for ( unsigned int j = 0; j < gridheight; j++ )
-      {
-        default_regionrealms[realm][i][j] = regionrealms[realm][i][j];
-      }
-    }
+    Range2d area = Range2d( Pos2d( 0, 0 ), XyToZone( realm->area().se() ), nullptr );
+    for ( const auto& p : area )
```

Comment:
The only issue here is that Range2d iterates over x first, right? So memory access for the zones could be optimized. I guess it depends how often this access happens